Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@sveltejs/kit' on production build

**Describe the bug**
When I create a new SvelteKit project with the `node` adapter and try to use the files in the `build` directory, I get the following error:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@sveltejs/kit' imported from [...]
```

**Logs**
```
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@sveltejs/kit' imported from [REDACTED]/app.js
    at packageResolve (internal/modules/esm/resolve.js:655:9)
    at moduleResolve (internal/modules/esm/resolve.js:696:18)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:810:11)
    at Loader.resolve (internal/modules/esm/loader.js:86:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:230:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:56:40)
    at link (internal/modules/esm/module_job.js:55:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**To Reproduce**
- create a new SvelteKit skeleton project
- add `adapter-node` to it
- make sure all the deps have `"next"` as version
- `yarn && yarn build`
- isolate the `build` directory (I only want the production files in my production environment) by moving it somewhere else
- add a `package.json` file to the directory with the contents: `{ "type": "module" }` (for ESM)
- `node .` <- This is what triggers the bug

**Expected behavior**

I expect the `build` directory to contain *everything* that is needed to run the app in isolation. Adding the small `package.json` file is another thing that I do not want to do myself.

When running `node .` the server should just run.

**Information about your SvelteKit Installation:**

<details>
  <summary>Diagnostics</summary>

```
  System:
    OS: macOS 11.2.3
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 5.37 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.16.0 - ~/.volta/tools/image/node/14.16.0/bin/node
    Yarn: 1.22.10 - ~/.volta/tools/image/yarn/1.22.10/bin/yarn
    npm: 7.7.0 - ~/.volta/tools/image/npm/7.7.0/bin/npm
  Browsers:
    Chrome: 90.0.4430.85
    Firefox: 87.0
    Safari: 14.0.3
  npmPackages:
    @sveltejs/kit: ^1.0.0-next.74 => 1.0.0-next.91 
    svelte: ^3.29.0 => 3.37.0 
    vite: ^2.2.2 => 2.2.3 
```

- Your adapter: Node
</details>

**Severity**
It's kind of annoying as I really want my production setup to be ready, but it doesn't stop me from development.

**Additional context**
All the above steps are done in a Dockerfile for production but I was able to reproduce it outside as well

<details>
  <summary>Dockerfile</summary>

```
FROM node:14-alpine as build

ARG VITE_API_URL

RUN mkdir /svelte-build
WORKDIR /svelte-build

ENV VITE_API_URL=${VITE_API_URL}

COPY . .
RUN yarn && yarn build

FROM node:14-alpine as production

RUN addgroup -S app && adduser -S app -G app

RUN mkdir /app && chown app:app /app

USER app

WORKDIR /app

RUN echo '{ "type": "module" }' > /app/package.json
COPY --chown=app:app --from=build /svelte-build/build/ /app

EXPOSE 3000

CMD node index.js
```
</details>

Same issue with me. Trying deploying to heroku using `adapter-node`
complete log:
<pre>
2021-05-26T13:05:39.000000+00:00 app[api]: Build succeeded
2021-05-26T13:05:41.632730+00:00 heroku[web.1]: Starting process with command `npm start`
2021-05-26T13:05:48.381759+00:00 app[web.1]:
2021-05-26T13:05:48.381783+00:00 app[web.1]: > mbaw@0.0.1 start /app
2021-05-26T13:05:48.381784+00:00 app[web.1]: > node build
2021-05-26T13:05:48.381784+00:00 app[web.1]:
2021-05-26T13:05:48.727284+00:00 app[web.1]: internal/process/esm_loader.js:74
2021-05-26T13:05:48.727483+00:00 app[web.1]: internalBinding('errors').triggerUncaughtException(
2021-05-26T13:05:48.727485+00:00 app[web.1]: ^
2021-05-26T13:05:48.727485+00:00 app[web.1]:
2021-05-26T13:05:48.727486+00:00 app[web.1]: Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@sveltejs/kit' imported from /app/build/app.js
2021-05-26T13:05:48.727487+00:00 app[web.1]: at packageResolve (internal/modules/esm/resolve.js:655:9)
2021-05-26T13:05:48.727487+00:00 app[web.1]: at moduleResolve (internal/modules/esm/resolve.js:696:18)
2021-05-26T13:05:48.727488+00:00 app[web.1]: at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:810:11)
2021-05-26T13:05:48.727488+00:00 app[web.1]: at Loader.resolve (internal/modules/esm/loader.js:88:40)
2021-05-26T13:05:48.727489+00:00 app[web.1]: at Loader.getModuleJob (internal/modules/esm/loader.js:241:28)
2021-05-26T13:05:48.727489+00:00 app[web.1]: at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:56:40)
2021-05-26T13:05:48.727490+00:00 app[web.1]: at link (internal/modules/esm/module_job.js:55:36) {
2021-05-26T13:05:48.727490+00:00 app[web.1]: code: 'ERR_MODULE_NOT_FOUND'
2021-05-26T13:05:48.727491+00:00 app[web.1]: }
2021-05-26T13:05:48.783848+00:00 app[web.1]: npm ERR! code ELIFECYCLE
2021-05-26T13:05:48.783850+00:00 app[web.1]: npm ERR! errno 1
2021-05-26T13:05:48.799210+00:00 app[web.1]: npm ERR! mbaw@0.0.1 start: `node build`
2021-05-26T13:05:48.799211+00:00 app[web.1]: npm ERR! Exit status 1
2021-05-26T13:05:48.799212+00:00 app[web.1]: npm ERR!
2021-05-26T13:05:48.799213+00:00 app[web.1]: npm ERR! Failed at the mbaw@0.0.1 start script.
2021-05-26T13:05:48.799214+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2021-05-26T13:05:49.642211+00:00 app[web.1]:
2021-05-26T13:05:49.654457+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
2021-05-26T13:05:49.654641+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2021-05-26T13_05_48_797Z-debug.log
2021-05-26T13:05:49.805437+00:00 heroku[web.1]: Process exited with status 1
2021-05-26T13:05:49.911449+00:00 heroku[web.1]: State changed from starting to crashed
2021-05-26T13:14:45.371410+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=mbaw.herokuapp.com request_id=eea42cf2-42a7-4ebf-a32d-65d92f4f638b fwd="187.8.172.13" dyno= connect= service= status=503 bytes= protocol=https
2021-05-26T13:14:45.904804+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=mbaw.herokuapp.com request_id=1319728b-26fb-49c9-bd22-dadf50c29172 fwd="187.8.172.13" dyno= connect= service= status=503 bytes= protocol=https
</pre>

--- a/.changeset/tiny-socks-sparkle.md
+++ b/.changeset/tiny-socks-sparkle.md
@@ -1,0 +1,5 @@
+---
+'@sveltejs/adapter-node': patch
+---
+
+Bundle server-side app during adapt phase

--- a/packages/adapter-node/index.js
+++ b/packages/adapter-node/index.js
@@ -1,6 +1,7 @@
-import { copyFileSync } from 'fs';
+import { readFileSync } from 'fs';
 import { join } from 'path';
 import { fileURLToPath } from 'url';
+import esbuild from 'esbuild';
 
 /**
  * @param {{
@@ -18,11 +19,17 @@ export default function ({ out = 'build' } = {}) {
 			utils.copy_client_files(static_directory);
 			utils.copy_static_files(static_directory);
 
-			utils.log.minor('Copying server');
-			utils.copy_server_files(out);
-
+			utils.log.minor('Building server');
 			const files = fileURLToPath(new URL('./files', import.meta.url));
-			copyFileSync(`${files}/server.js`, `${out}/index.js`);
+			utils.copy(files, '.svelte-kit/node');
+			await esbuild.build({
+				entryPoints: ['.svelte-kit/node/index.js'],
+				outfile: join(out, 'index.js'),
+				bundle: true,
+				external: Object.keys(JSON.parse(readFileSync('package.json', 'utf8')).dependencies || {}),
+				format: 'esm',
+				platform: 'node'
+			});
 
 			utils.log.minor('Prerendering static pages');
 			await utils.prerender({

--- a/packages/adapter-node/package.json
+++ b/packages/adapter-node/package.json
@@ -20,6 +20,9 @@
 		"check-format": "prettier --check . --config ../../.prettierrc --ignore-path .gitignore",
 		"prepublishOnly": "npm run build"
 	},
+	"dependencies": {
+		"esbuild": "^0.12.5"
+	},
 	"devDependencies": {
 		"@rollup/plugin-json": "^4.1.0",
 		"@sveltejs/kit": "workspace:*",

--- a/packages/adapter-node/rollup.config.js
+++ b/packages/adapter-node/rollup.config.js
@@ -5,10 +5,10 @@ import json from '@rollup/plugin-json';
 export default {
 	input: 'src/index.js',
 	output: {
-		file: 'files/server.js',
+		file: 'files/index.js',
 		format: 'esm',
 		sourcemap: true
 	},
 	plugins: [nodeResolve(), commonjs(), json()],
-	external: ['./app.js', ...require('module').builtinModules]
+	external: ['../output/server/app.js', ...require('module').builtinModules]
 };

--- a/packages/adapter-node/src/index.js
+++ b/packages/adapter-node/src/index.js
@@ -1,3 +1,4 @@
+import './require_shim';
 import { createServer } from './server';
 // TODO hardcoding the relative location makes this brittle
 import { render } from '../output/server/app.js'; // eslint-disable-line import/no-unresolved

--- a/packages/adapter-node/src/index.js
+++ b/packages/adapter-node/src/index.js
@@ -1,10 +1,10 @@
 import { createServer } from './server';
-/*eslint import/no-unresolved: [2, { ignore: ['\.\/app\.js$'] }]*/
-import * as app from './app.js';
+// TODO hardcoding the relative location makes this brittle
+import { render } from '../output/server/app.js'; // eslint-disable-line import/no-unresolved
 
 const { HOST = '0.0.0.0', PORT = 3000 } = process.env;
 
-const instance = createServer({ render: app.render }).listen(PORT, HOST, () => {
+const instance = createServer({ render }).listen(PORT, HOST, () => {
 	console.log(`Listening on port ${PORT}`);
 });
 

--- a/packages/adapter-node/src/require_shim.js
+++ b/packages/adapter-node/src/require_shim.js
@@ -1,0 +1,2 @@
+import { createRequire } from 'module';
+global.require = createRequire(import.meta.url);

